pkg/storegateway/bucket.go

```diff
@@ -1344,6 +1351,13 @@ func (s *BucketStore) LabelNames(ctx context.Context, req *storepb.LabelNamesReq
 			continue
 		}
 
+		if b.meta.Compaction.Level == 1 && b.meta.Thanos.Source == block.ReceiveSource && !b.queriedNonCompactedBlock {
```

Comment:
Can we extract the "is non-compacted" check into a method?

Reply:
Removed this, so we're only logging in 1 place now.